lib/ddtrace/contrib/http/instrumentation.rb

```diff
@@ -28,8 +29,12 @@ def self.after_request(&block)
 
         # InstanceMethods - implementing instrumentation
         module InstanceMethods
+          include Datadog::Contrib::HttpAnnotationHelper
+
           def request(req, body = nil, &block) # :yield: +response+
-            pin = datadog_pin
+            host, = host_and_port(req)
+            request_options = datadog_configuration(host)
+            pin = datadog_pin(request_options)
             return super(req, body, &block) unless pin && pin.tracer
 
             if Datadog::Contrib::HTTP.should_skip_tracing?(req, @address, @port, pin.tracer)
```

Comment:
Something that's really cool about this resolver is we might be able to use it to remove this circuit breaker (which exists to prevent our tracer from tracing itself when it writes traces over HTTP to the agent.) Alternatively, we could probably use `enabled` as well to effect the same result.

Probably should be addressed in another PR though.